lib/rouge/lexers/janet.rb

```diff
@@ -187,8 +188,9 @@ def name_token(name)
       end
 
       state :quote do
-        rule %r/[\(\[]/, Punctuation, :push
-        rule symbol, Str::Symbol
+        rule %r/[\(\[\{]/, Punctuation, :push
+        rule %r/[\)\]\}]/, Punctuation, :pop!
```

Comment:
Backslashes shouldn't be necessary inside of `[]` except for `[]` themselves - it might be much easier to read that way.